### PR TITLE
Adds timeoutIntervals, which controls the variable in BasicTransport

### PR DESCRIPTION
**Pull request by OferMania**
NF-362

### CODE REVIEW

File: patches/transport-timeout-interval.patch [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
++#include "OptionParser.h"
+ #include "Service.h"
+ #include "TimeTrace.h"
+ #include "WorkerManager.h"
+@@ -110,6 +111,10 @@ BasicTransport::BasicTransport(Context* context, const ServiceLocator* locator,
+     timerInterval = Cycles::fromMicroseconds(2000);
+     nextTimeoutCheck = Cycles::rdtsc() + timerInterval;
+ 
++    if (context->options != NULL) {
++        timeoutIntervals = context->options->getTimeoutIntervals();
```
Comment (vexingcodes) [inline patches/transport-timeout-interval.patch:30]: `if (context->options->getTimeoutIntervals() != 0)` should probably go here. If somebody doesn't specify this option, then it defaults to `0`. If we use the value of `0` and overwrite the RAMCloud hardcoded default then that's bad and I think that will cause a crash later due to some assertions RAMCloud makes. At any rate, it seems best to use the hardcoded default if `getTimeoutIntervals()` returns `0`. Your comments in `OptionParser.cc` about the default being `40` also indicate that this should be done.

**Review by vexingcodes — CHANGES_REQUESTED** [review]
Looks good. We need to make sure the default gets applied correctly though.

File: patches/transport-timeout-interval.patch (hunk 1 above)
Comment (OferMania) [reply patches/transport-timeout-interval.patch:30]: Modified here. Also set default in OptionParser.cc to zero for consistency w other arguments. I verified by hand that, when timeoutIntervals argument isn't specified, we DO use the default value of 40 due to default constructor value population in BasicTransport.cc. We NEVER actually want timeoutIntervals to be zero in BasicTransport.cc, and anyhow, there's an assert line further down in this same file that effectively requires timeoutIntervals to always be higher than 2 * pingIntervals (which is hardcoded to 3, hence timeoutIntervals must always be higher than 6)

**Review by vexingcodes — APPROVED** [review]
Looks good.

**Pull request merged**